### ISSUE TITLE
Custom Profile field for any company appears for all the companies

### ISSUE THREAD

**Issue by syedmahtab** [iomad/iomad#1264]
We plan to create custom fields specific to different companies. But, when we create the fields, they appear for all the companies. Is this not an issue and how to resolve this?
Thanks,

**Comment by syedmahtab** [comment]
Hi,
Scenario:
I have three companies setup in iomad (e.g. C1, C2, C3). I setup custom profile fields for C1 and C2. When I go to edit any user of any company, all the profile fields of C1 & C2 appear on Edit profile page.

While probing into the issue, in an attempt to resolve the same, I changed SQL query a little in /user/profile/lib.php under function profile_get_user_fields_with_data from:
        $sql .= " AND (uif.categoryid IN (
                  SELECT c.profileid FROM {company} c JOIN {company_users} cu ON (c.id = cu.companyid AND cu.userid = :companyuserid))
                  OR uif.categoryid IN (
                  SELECT id FROM {user_info_category} WHERE id NOT IN (SELECT profileid from {company}))) ";

To:
        $sql .= " WHERE (uif.categoryid IN (
                  SELECT c.profileid FROM {company} c JOIN {company_users} cu ON (c.id = cu.companyid AND cu.userid = :companyuserid))
                  OR uif.categoryid IN (
                  SELECT id FROM {user_info_category} WHERE id NOT IN (SELECT profileid from {company}))) ";

Basically, I changed 'AND' at the start of this query with a 'WHERE' and it solves the issue. I wanted to confirm with you guys if this is correct implementation. In case, this is correct, would it be fine if we may implement the change in core IOMAD code for next releases.

Thanks,
Mahtab Hussain

**Comment by turf212** [comment]
There was a fix for this applied in #1261 This was applied to latest 3.6 and 3.7 branches.  Can you please ensure you have the latest version of the code as there was also an issue fixed last week with regards to company profile fields allowing a space in the shortname.
